scripts/release_archive.py

```
@@ -0,0 +1,292 @@
+
+
+def cargo_program_and_args(cargo: str) -> tuple[str, list[str]]:
+    """Return the executable and wrapper arguments represented by `cargo`."""
+    stripped_cargo = cargo.strip()
+    if not stripped_cargo:
+        raise SystemExit("cargo executable cannot be empty")
+    if looks_like_executable_path(stripped_cargo):
+        return strip_matching_quotes(stripped_cargo), []
+    cargo_command = shlex.split(stripped_cargo)
+    if not cargo_command:
+        raise SystemExit("cargo executable cannot be empty")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)